File: Leetcode/38_CaS.js
/**
 * @param {number} n
 * @return {string}
 */
var countAndSay = function(n) {
    let result = "1";

    for (let i = 1; i < n; i++) {
        let nextStr = "";
        let count = 1;

        for (let j = 0; j < result.length; j++) {
            if (result[j] === result[j + 1]) {
                count++;
            } else {
                nextStr += count.toString() + result[j];
                count = 1;
            }
        }
        result = nextStr;
    }

    return result;
};

console.log(countAndSay(1));
console.log(countAndSay(2));
console.log(countAndSay(3));

File: Leetcode/38_CaS.py
class Solution:
    def countAndSay(self, n: int) -> str:
        current = "1"
        for _ in range(n-1):
            next_str = []
            i = 0

            while i < len(current):
                count = 1
                while i + 1 < len(current) and current[i] == current[i+1]:
                    count += 1
                    i += 1
                
                next_str.append(str(count))
                next_str.append(current[i])
                i+=1
            current ="".join(next_str)
        return current
print(Solution().countAndSay(1))  # Output: "1"
print(Solution().countAndSay(4))  # Output: "1211"